### PR TITLE
Generate single manifest for installation #(228)

### LINKED ISSUE #228: 1-step installation
**Is your feature request related to a problem? Please describe.**
Our current [installation guide](https://www.rabbitmq.com/kubernetes/operator/install-operator.html#using-kubectl-yaml) has a number of steps that are not "beginner friendly", for example building your own Docker image or clonning the repo. These steps assume developer experience with Git and with Docker, which might not be true for some users of this project, who might have a more "IT-ops" kind of background. We should make the installation experience as simple as possible.

**Describe the solution you'd like**
Inspired by project [Antrea](https://github.com/vmware-tanzu/antrea/blob/master/docs/getting-started.md#installation), we should aim to have an installation method as simple as `kubectl apply -f <github-url/to/a/file.yaml>`. Kubectl already supports pointing to URLs and it has a built-in module for `kustomize` files, which can be invoked using `kubectl apply -k <github-url>//path/to/some/folder/`. We might need to create a new kustomization folder, or change the current layout (dangerous) to achieve this.

Examples of the above working for this repo. Please note that the double forward slash is intended and required:

```bash
kubectl create -k github.com/rabbitmq/cluster-operator//config/namespace/
kubectl create -k github.com/rabbitmq/cluster-operator//config/crd/
```

**Describe alternatives you've considered**
Using pure `kustomize` is also posible e.g. `kustomize build github.com/rabbitmq/cluster-operator//config/rbac/`, however the adoption and use of `kustomize` is not guaranteed and some users might not have this tool installed. It is almost guaranteed that a Kubernetes user will have `kubectl` installed, therefore we should favour the use `kubectl` when possible.

**Additional context**
We can't invoke `kubectl create -k github.com/rabbitmq/cluster-operator//config/default/base/` because it will try to load files from `../some/other/location`, which won't exist if we use this method..

Git should be aware of symlinks, it's worth trying out and exploring if a "central" folder with symlinks to other resources can be a solution.

### PR DESCRIPTION
**Pull request by harshac**
This relates #228 

**Note to reviewers:** remember to look at the commits in this PR and consider if they can be squashed

## Summary Of Changes

Adapted `generate-installation-manifest` make target to generate a single-file manifest with all the resources required to deploy Cluster Operator. The image name and tag is hard-coded, however it can be (and it's expected to) changed using `kustomize edit set image <..>` command. The pipeline will run a similar `kustomize` command to generate release specific, versioned, manifest files.

## Additional Context

This change is the first step to accomplish 1-step install. Second step is to adapt the pipeline to leverage this change and produce versioned manifest files.

## Local Testing

N/A no changes to code or to how we deploy.

Edit: @Zerpet picked this PR and added some changes.

### CODE REVIEW

File: config/installation/kustomization.yaml [hunk 1]
```diff
@@ -7,10 +7,16 @@
...
 namespace: rabbitmq-system
+images:
+- name: controller
+  newName: rabbitmqoperator/rabbitmq-cluster-kubernetes-operator-dev
+  newTag: latest
 
-patches:
-- manager_deployment_patch.yaml
+resources:
+- generated/namespace.yaml
```
Comment (harshac) [inline config/installation/kustomization.yaml:19]: Check if we can directly use the sibling directory resource here for namespace and crd

File: Makefile [hunk 2]
```diff
@@ -90,19 +90,22 @@ deploy-kind: check-env-docker-repo git-commit-sha patch-kind manifests deploy-na
...
 	kustomize build config/namespace/base/ > charts/operator/templates/namespace.yaml
 	kustomize build config/crd/ > charts/operator/templates/crd.yaml
 	kustomize build config/rbac/ > charts/operator/templates/rbac.yaml
 	kustomize build config/default/overlays/helm/ > charts/operator/templates/deployment.yaml
 
+install_dir = config/installation/generated
+generate-installation-manifest:
+	mkdir -p $(install_dir)
+	mkdir -p releases
+	cp config/namespace/base/namespace.yaml $(install_dir)/namespace.yaml
```
Comment (harshac) [inline Makefile:103]: Remove the cp if we can use sibling resources in kustomization.yml under config/installation

File: Makefile [hunk 3]
```diff
@@ -90,19 +90,22 @@ deploy-kind: check-env-docker-repo git-commit-sha patch-kind manifests deploy-na
...
-	kustomize build config/installation > installation/operator.yaml
-
 generate-helm-manifests:
 	kustomize build config/namespace/base/ > charts/operator/templates/namespace.yaml
 	kustomize build config/crd/ > charts/operator/templates/crd.yaml
 	kustomize build config/rbac/ > charts/operator/templates/rbac.yaml
 	kustomize build config/default/overlays/helm/ > charts/operator/templates/deployment.yaml
 
+install_dir = config/installation/generated
+generate-installation-manifest:
```
Comment (harshac) [inline Makefile:100]: Should accept a docker-image-url, if missing points to latest.

File: Makefile [hunk 4]
```diff
@@ -90,19 +90,22 @@ deploy-kind: check-env-docker-repo git-commit-sha patch-kind manifests deploy-na
...
+	mkdir -p $(install_dir)
+	mkdir -p releases
+	cp config/namespace/base/namespace.yaml $(install_dir)/namespace.yaml
+	cp config/crd/bases/rabbitmq.com_rabbitmqclusters.yaml $(install_dir)/rabbitmq.com_rabbitmqclusters.yaml
+	kustomize build config/rbac/ > $(install_dir)/rbac.yaml
+	kustomize build config/manager/ > $(install_dir)/manager.yaml
+	kustomize build config/installation/ > releases/rabbitmq-cluster-operator.yaml
```
Comment (harshac) [inline Makefile:107]: Replace the docker image url before building the rabbitmq-cluser-operator.yaml if provided.

File: config/installation/kustomization.yaml [hunk 5]
```diff
@@ -7,10 +7,16 @@
...
 
-resources:
-- ../default/base
-
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
 namespace: rabbitmq-system
+images:
+- name: controller
+  newName: rabbitmqoperator/rabbitmq-cluster-kubernetes-operator-dev
```
Comment (harshac) [inline config/installation/kustomization.yaml:15]: Change the image name to match the dockerhub image

**Review by harshac — COMMENTED** [review]
* Also figure out how to override the docker image url without cloning the repo in case of private registry
* Modify the docs

**Review by coro — APPROVED** [review]

**Review by mkuratczyk — APPROVED** [review]
Looks good!

**Comment by Zerpet** [comment]
Checks are green 🟢 and with two approvals ✅ I'm merging this PR 🚀

**Pull request merged**